### PR TITLE
Implement setup password form - Closes #4293

### LINKED ISSUE #4293: Implement setup password form
### Description
Enable users to setup password for their accounts

### Acceptance Criteria

- Implement setup your password form according to [design](https://www.figma.com/file/GbrsJiZlsh2OzVKqYDD4zI/Accountmanaging-improvements?node-id=139%3A10414)
- Implement password validation and confirm your password field  as mentioned in LIP - Key encryption
[design](https://www.figma.com/file/GbrsJiZlsh2OzVKqYDD4zI/Account-managing-improvements?node-id=149%3A16183)
- Implement validate account name, min and max length of name (2-50)
- Save account is disabled while User does not agreed to store my encrypted
- Props (onSubmit(encryptedSchema), onSkip(), recoverPhrase)
- Password field should have toggle show/hide button
- submit button its disabled until the password is filled and user approved agrement
- Allow user to enter the custom name but for delegate it should be auto filled from user-name delegate and it cant be change
- Add unit tests to 
     - Validate password and confirm your password
     - Setup password with entering valid password
     - Setup password with entering invalid confirm your password

### Additional Information
- Component will receive the recovery phrase as props and it should return backup json schema on the submit
- The logic of encryption should be on a custom hook 
- Implement setup password success dialog

### PR DESCRIPTION
**Pull request by eniolam1000752**
### What was the problem?

This PR resolves #4293

### How was it solved?

- [x] Adapt the input component to be used as a password component
- [x] Construct form using `react-hook-form`
- [x] Create validation schemas for the password form 
- [x] Write unit test

### How was it tested?

- Visually
- Writing jest tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -164,6 +165,7 @@
     "lottie-web": "5.5.5",
     "moment": "2.23.0",
     "node-hid": "2.1.1",
+    "npm": "8.10.0",
```
Comment (soroushm) [inline package.json]: remove npm `install` and `npm` packages

File: src/modules/auth/components/SetPasswordForm/SetPasswordForm.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  return (
+    <div>
+      <div className={`${styles.titleHolder} ${grid['col-xs-12']}`}>
+        <h1>{t('Set up device password')}</h1>
+        <p>
+          {t('This password is used to encrypt your secret recovery phrase, which will be used for managing your account.')}
+        </p>
+      </div>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <div style={{ padding: '0px 30px' }}>
```
Comment (isalga) [inline src/modules/auth/components/SetPasswordForm/SetPasswordForm.js]: Put this styles in css please.

File: src/modules/auth/components/SetPasswordForm/SetPasswordForm.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+          <Input
+            size="xs"
+            secureTextEntry
+            feedback={errors.password?.message}
+            status={errors.password ? 'error' : undefined}
+            defaultValues=""
+            value={password}
+            label={(
+              <span style={{ display: 'flex', alignItems: 'center' }}>
```
Comment (isalga) [inline src/modules/auth/components/SetPasswordForm/SetPasswordForm.js]: same

**Review by isalga — CHANGES_REQUESTED** [review]

File: setup/config/webpack.config.dev.js [hunk 4]
```diff
@@ -23,7 +23,7 @@ const config = {
     new ESLintPlugin({
       context: '../',
     }),
-    new webpack.IgnorePlugin({ resourceRegExp: /\.\/locale$/ }),
+    // new webpack.IgnorePlugin({ resourceRegExp: /\.\/locale$/ }),
```
Comment (soroushm) [inline setup/config/webpack.config.dev.js:26]: Hi @reyraa,
we had difficulty here .local file was ignored, I believe it had issues at the moment, but now it looks working fine do you know about it?

**Review by soroushm — DISMISSED** [review]

**Review by ikem-legend — APPROVED** [review]

**Review by soroushm — APPROVED** [review]

**Pull request merged**